diff --git a/drivers/usb/gadget/ether.c b/drivers/usb/gadget/ether.c
index 8f03a6b..21a38b9 100644
--- a/drivers/usb/gadget/ether.c
+++ b/drivers/usb/gadget/ether.c
@@ -90,7 +90,7 @@ static const char driver_desc[] = DRIVER_DESC;
 			|USB_CDC_PACKET_TYPE_PROMISCUOUS \
 			|USB_CDC_PACKET_TYPE_DIRECTED)
 
-#define USB_CONNECT_TIMEOUT (3 * CONFIG_SYS_HZ)
+#define USB_CONNECT_TIMEOUT (15 * CONFIG_SYS_HZ)
 
 /*-------------------------------------------------------------------------*/
 
@@ -235,8 +235,13 @@ static inline int BITRATE(struct usb_gadget *g)
  * used with CDC Ethernet, Linux 2.4 hosts will need updates to choose
  * the non-RNDIS configuration.
  */
+#ifdef CONFIG_SPL_BUILD
 #define RNDIS_VENDOR_NUM	0x0525	/* NetChip */
 #define RNDIS_PRODUCT_NUM	0xa4a2	/* Ethernet/RNDIS Gadget */
+#else
+#define RNDIS_VENDOR_NUM    0x0525
+#define RNDIS_PRODUCT_NUM   0xa4a5
+#endif
 
 /*
  * Some systems will want different product identifers published in the
diff --git a/include/configs/am335x_evm.h b/include/configs/am335x_evm.h
index a87059c..ca3547a 100644
--- a/include/configs/am335x_evm.h
+++ b/include/configs/am335x_evm.h
@@ -184,13 +184,25 @@
 	DFUARGS
 #endif
 
+#undef CONFIG_CMD_DHCP
+#undef CONFIG_DRIVER_TI_CPSW
+#undef CONFIG_SPL_ETH_SUPPORT
+
+#define CONFIG_ENV_IS_NOWHERE       /* Store ENV in memory only */
+#define CONFIG_SILENT_CONSOLE
+
+#define CONFIG_AUTOBOOT_DELAY_STR		"\x1B"	/* ESC */
+#define CONFIG_AUTOBOOT_STOP_STR		"\x20" /* SPACE */
+#define CONFIG_MENUKEY				    27
+
+#define CONFIG_ETHADDR			00:ff:ff:ff:ff:00
+
 #define CONFIG_BOOTCOMMAND \
-	"run findfdt; " \
-	"run mmcboot;" \
-	"setenv mmcdev 1; " \
-	"setenv bootpart 1:2; " \
-	"run mmcboot;" \
-	"run nandboot;"
+    "setenv ethact usb_ether;" \
+    "setenv serverip 192.168.1.9;" \
+    "setenv ipaddr 192.168.1.3;" \
+    "tftp 0x82000000 itb;" \
+    "bootm;"
 
 /* NS16550 Configuration */
 #define CONFIG_SYS_NS16550_COM1		0x44e09000	/* Base EVM has UART0 */
diff --git a/include/configs/ti_armv7_common.h b/include/configs/ti_armv7_common.h
index 110a4f8..cd33076 100644
--- a/include/configs/ti_armv7_common.h
+++ b/include/configs/ti_armv7_common.h
@@ -54,16 +54,16 @@
 #define DEFAULT_LINUX_BOOT_ENV \
 	"loadaddr=0x82000000\0" \
 	"kernel_addr_r=0x82000000\0" \
-	"fdtaddr=0x88000000\0" \
+	"fdtaddr=0x80F80000\0" \
 	"fdt_addr_r=0x88000000\0" \
-	"rdaddr=0x88080000\0" \
+	"rdaddr=0x81000000\0" \
 	"ramdisk_addr_r=0x88080000\0" \
 	"bootm_size=0x10000000\0"
 
 /*
  * Default to a quick boot delay.
  */
-#define CONFIG_BOOTDELAY		1
+#define CONFIG_BOOTDELAY		0
 
 /*
  * DDR information.  If the CONFIG_NR_DRAM_BANKS is not defined,
@@ -130,10 +130,10 @@
 #ifdef CONFIG_DFU_MMC
 #define CONFIG_SYS_MALLOC_LEN	((16 << 20) + CONFIG_SYS_DFU_DATA_BUF_SIZE)
 #else
-#define CONFIG_SYS_MALLOC_LEN	(16 << 20)
+#define CONFIG_SYS_MALLOC_LEN	(1024 << 10)
 #endif
 #define CONFIG_SYS_HUSH_PARSER
-#define CONFIG_SYS_PROMPT		"U-Boot# "
+#define CONFIG_SYS_PROMPT		"USB-Flash#"
 #define CONFIG_SYS_CONSOLE_INFO_QUIET
 #define CONFIG_BAUDRATE			115200
 #define CONFIG_ENV_VARS_UBOOT_CONFIG	/* Strongly encouraged */
